public/app/features/explore/state/query.ts

```diff
@@ -264,8 +264,8 @@ export const importQueries = (
     let importedQueries = queries;
     // Check if queries can be imported from previously selected datasource
     if (sourceDataSource.meta?.id === targetDataSource.meta?.id) {
-      // Keep same queries if same type of datasource
-      importedQueries = [...queries];
+      // Keep same queries if same type of datasource, but delete datasource query property to prevent miss-match of new and old data source instance
```

Comment:
```suggestion
      // Keep same queries if same type of datasource, but delete datasource query property to prevent mismatch of new and old data source instance
```